src/clims/models/assignment.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+
+from django.db import models
+from sentry.db.models import (FlexibleForeignKey, sane_repr)
+from jsonfield import JSONField
+
+
+class SubstanceAssignment(models.Model):
+    """
+    Represents an assignment of a substance to a process.
+
+    Note that this model only represents assignments by humans to a process. The process engine
+    will moves the entity around after this assignment without it being logged here.
+    """
+
+    STATUS_NONE = 0
+
+    # A request is about to be sent to the external workflow engine
+    STATUS_REQUESTING = 1
+
+    # A request has been delivered to the external workflow engine. The execution is now under
+    # the workflow engine's control
+    STATUS_DELIVERED = 2
+
+    __core__ = True
+
+    substance = FlexibleForeignKey('clims.Substance', related_name="assignments")
+
+    user = FlexibleForeignKey('sentry.User')
+
+    workflow = FlexibleForeignKey('clims.Workflow', null=True)
+
+    # The preset of variables used when assigning
+    preset = models.TextField()
```

Comment:
Presets can be thought of as a named tuple of `(process, variables)`. They are only there for the end user, so they don't have to specify variables that almost always might be the same.

It's similar to having a function for end users of a library that calls a lower level function that does all the work, but is too inticrate for end users, e.g.:

```
def _print(self, a, b, c, color, e, f, g, **kwargs): ...
 
def print_color(self): ...

def print_bw(self): ...
```

This variable in the model is just to know which preset was selected (the name of it) so it can be shown in the UI.

Reply:
Yes, I think this became clearer to me as well, as I read more of the code.